### PR TITLE
More code clean up: avoid wild card imports, avoid mutable defaults, check empty list in pythonic way

### PR DESCRIPTION
**Pull request by hwpang**

### CODE REVIEW

File: pysidt/extensions.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def get_extension_edge(
 
     iter = 0
 
-    while grps[iter] != []:
```
Comment (mjohnson541) [inline pysidt/extensions.py]: Are we sure grps[iter] can never be None?

**Review by mjohnson541 — COMMENTED** [review]
Looks mostly good, just one question.

File: pysidt/extensions.py (hunk 1 above)
Comment (hwpang) [reply pysidt/extensions.py]: `while None` also counts as `False`

Comment (hwpang) [reply pysidt/extensions.py]:
```
>>> bool(None)
False
```

Comment (mjohnson541) [reply pysidt/extensions.py]: Yes, which mean that if None => False, but if None != [] => True. So there would be different behavior.

Comment (mjohnson541) [reply pysidt/extensions.py]: I've seen that modification create new bugs in other situations.

Comment (hwpang) [reply pysidt/extensions.py]: hmmm interesting. Ok I will change it back.

**Comment by hwpang** [comment]
@mjohnson541 This is ready for another review

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**